### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by BYK**
Fixed Model.save() raising exception when nothing is changed in
object.
Fixed Model.save() does not return anything on success.
Made validators.Regex usable via constructor.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -29,65 +29,65 @@ class ModelBase(type):
...
+
         # See cursor.description
         # http://www.python.org/dev/peps/pep-0249/
         if not hasattr(new_class, "db"):
             new_class.db = autumn_db
-            
+
         db = new_class.db
-        q = Query.raw_sql('SELECT * FROM %s LIMIT 1' % new_class.Meta.table_safe, db=new_class.db)
+        q = Query.raw_sql('SELECT * FROM %s LIMIT 1' % new_class.Meta.table_safe, db = new_class.db)
```
Comment (bondgeek) [inline model.py]: I don't think this matches the preferred Python style guidelines, see PEP 8:
"- Don't use spaces around the '=' sign when used to indicate a
      keyword argument or a default parameter value.
"

Comment (BYK) [inline model.py]: Yup, sorry about all those differences. Due to automatic code formatting of PyDev. Fixing right away...

File: model.py [hunk 2]
```diff
@@ -198,41 +198,45 @@ class Model(object):
...
         query += ' WHERE %s = %s ' % (escape(self.Meta.pk), self.db.conn.placeholder)
-        
+
         values = [getattr(self, f) for f in self._changed]
         values.append(self._get_pk())
-        
+
         cursor = Query.raw_sql(query, values, self.db)
-        
+        return True
```
Comment (bondgeek) [inline model.py:215]: This isn't the behavior I had in mind for Model.save()  (I think LuckyTheTourist, nee SuperJared, also envisioned Model.save() returning True only if a new record was created)   But I don't remember why that seemed like a good idea or how much if any I'm relying on that behavior.  I'll do a quick review.

Comment (BYK) [inline model.py:215]: Okay. I ıntroduced this after fixing the failing _update method when nothing is changed. It now returns False if no update is done though I do not rely on this return value also so we can change it.

Comment (bondgeek) [inline model.py:215]: I'm leaving it as 'True'.   I'm not relying on it--seems cleaner to have the logic for checking if a record is new or an update before committing by calling save() anyway.

**Pull request merged**